application/comit_node/src/swap_protocols/rfc003/ethereum/htlc_events.rs

```diff
@@ -0,0 +1,223 @@
+use crate::{
+    ledger_query_service::{EthereumQuery, EventMatcher, QueryEthereum, Topic},
+    swap_protocols::{
+        asset::Asset,
+        ledger::Ethereum,
+        rfc003::{
+            self,
+            ethereum::{extract_secret::extract_secret, REDEEM_LOG_MSG, REFUND_LOG_MSG},
+            events::{Deployed, Funded, HtlcEvents, RedeemedOrRefunded},
+            state_machine::HtlcParams,
+            FundTransaction, RedeemTransaction, RefundTransaction,
+        },
+    },
+};
+use ethereum_support::{
+    web3::types::Address, CalculateContractAddress, Erc20Token, EtherQuantity, Transaction,
+};
+use futures::{
+    future::{self, Either},
+    Future,
+};
+use std::sync::Arc;
+
+impl HtlcEvents<Ethereum, EtherQuantity> for Arc<dyn QueryEthereum + Send + Sync + 'static> {
+    fn htlc_deployed(
+        &self,
+        htlc_params: HtlcParams<Ethereum, EtherQuantity>,
+    ) -> Box<Deployed<Ethereum>> {
+        let query_ethereum = Arc::clone(&self);
+        let query = query_ethereum
+            .create(EthereumQuery::Transaction {
```

Comment:
Create method on EthereumQuery to create a contract creation query given the bytecode.